src/execution_tree/primitives/shuffle_operation.cpp

```diff
@@ -44,6 +44,8 @@ namespace phylanx { namespace execution_tree { namespace primitives
     };
 
     ///////////////////////////////////////////////////////////////////////////
+    std::mt19937 shuffle_operation::rand_machine{std::random_device{}()};
```

Comment:
You're missing `#include <random>` for this file.

Reply:
It's in the header file.